packages/cardpay-cli/utils.ts

```diff
@@ -4,11 +4,19 @@ import { AbstractProvider } from 'web3-core';
 import { HttpProvider, getConstant, networkIds, getConstantByNetwork } from '@cardstack/cardpay-sdk';
 import WalletConnectProvider from '@walletconnect/web3-provider';
 import { Options } from 'yargs';
+/* eslint-disable @typescript-eslint/no-require-imports,@typescript-eslint/no-var-requires */
+const TrezorWalletProvider = require('trezor-cli-wallet-provider');
 
 const BRIDGE = 'https://safe-walletconnect.gnosis.io/';
 
-export async function getWeb3(network: string, mnemonic?: string): Promise<Web3> {
-  if (mnemonic) {
+export async function getWeb3(network: string, mnemonic?: string, trezor?: boolean): Promise<Web3> {
+  if (trezor) {
```

Comment:
Then in our connection type choosing logic we can use a really cool typescript pattern that will fail type checking if we decide to add a new connection type without adding a case statement to support it:
```ts
switch(connectionType) {
  case 'wallet-connect': 
    .
    .
  case 'mnemonic':
    .
    .
  case 'trezor': 
    .
    .
  default:
    assertNever(connectionType);
}

function assertNever(value: never) {
  throw new Error(`should never happen ${value}`);
}
```